js/src/Builder.js

```diff
@@ -26,6 +27,12 @@ var d3_select = require('d3-selection').select
 var d3_selection = require('d3-selection').selection
 var d3_json = require('d3-request').json
 
+// stuff for difference mode
+var difference_mode_active
```

Comment:
These should be defined as parameters of a Builder instance. i.e. `this.difference_mode_active = false` in the init function below.